src/detectors/typeChecker/FileLinter.ts

```diff
@@ -207,6 +208,97 @@ export default class FileLinter {
 		});
 	}
 
+	/**
+	 * Gets arguments from a CallExpression.
+	 *
+	 * It could be the N-th argument from the call or it could
+	 * find a named argument within argument's list.
+	 * The callable function should be a method of an instance i.e.
+	 * Instance.fnInvoke();
+	 *
+	 * @param {ts.CallExpression} node
+	 * @param {object} settings
+	 * @param {string} settings.callExpressionName Method's name
+	 * @param {string} settings.callExpressionPropName Object's instance name
+	 * @param {string|number} settings.fnArgument The N-th argument or argument's name
+	 * @returns
+	 */
+	getFnArgument(node: ts.CallExpression, settings: {
+		callExpressionName: string;
+		callExpressionPropName: string;
+		fnArgument: string | number; // Get N-th argument or argument by its name
+	}) {
+		const nodeExp = node.expression as ts.PropertyAccessExpression;
+		if (nodeExp?.name?.text !== settings.callExpressionName ||
+			!ts.isPropertyAccessExpression(nodeExp) ||
+			!ts.isIdentifier(nodeExp.expression) ||
+			nodeExp?.expression?.text !== settings.callExpressionPropName) {
+			// Didn't match Instance.call(), so we're not interested anymore of analyzing it.
+			return;
+		}
+
+		let fnArgumentNode;
+
+		if (typeof settings.fnArgument === "number") { // Get N-th arg
+			fnArgumentNode = node.arguments[settings.fnArgument];
+		} else {
+			// Arg by name
+			fnArgumentNode = node.arguments.find((arg: ts.Expression) =>
+				arg.kind === ts.SyntaxKind.Identifier && arg.getText() === settings.fnArgument);
+		}
+		return fnArgumentNode ?? null;
+	}
+
+	checkLibInitCall(node: ts.CallExpression) {
+		const nodeExp = node.expression as ts.PropertyAccessExpression;
+		const {symbol} = this.#checker.getTypeAtLocation(nodeExp);
+		// TS parser uses some intermediate types that are not available as definitions.
+		// In this case SymbolObject which is a ts.Symbol + ts.Node and that's
+		// why we need these ugly type castings
+		const importDeclaration =
+			((((symbol as unknown) as ts.Node)?.parent?.parent as unknown) as ts.Symbol)?.getEscapedName() as string;
```

Comment:
Is this really the only solution? So far we didn't need this in this file, so I wonder what's so special here.

Reply:
I don't think we have such a case already.

What it does is to go up the tree to the sap.ui.define/require call and check the name of the imported module.
So far we directly access the sap.ui.define/require and didn't need to go up the tree.
This code verifies that the local `Lib.init()` call is actually the `sap.ui.core.Lib.init()`.

I'd appreciate any suggestions to improve this one

Reply:
Done